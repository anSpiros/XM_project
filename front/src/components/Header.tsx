import React from "react";
import "../scss/header.scss";
import { NavLink } from "react-router-dom";
import { FORM_ROUTE, RESULTS_ROUTE, ROOT_ROUTE } from "../routes/routes";

function Header() {
  return (
    <div className="container-fluid header">
      <div className="row">
        <div className="col-12">
          <ul className="list-unstyled header-list mb-0">
            <li>
              <NavLink exact to={ROOT_ROUTE} activeClassName="active">
                Home
              </NavLink>
            </li>
            <li>
              <NavLink exact to={FORM_ROUTE} activeClassName="active">
                Form
              </NavLink>
            </li>
            <li>
              <NavLink exact to={RESULTS_ROUTE} activeClassName="active">
                Results
              </NavLink>
            </li>
          </ul>
        </div>
      </div>
    </div>
  );
}

export default Header;
