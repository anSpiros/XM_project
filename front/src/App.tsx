import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { FORM_ROUTE, RESULTS_ROUTE, ROOT_ROUTE } from "./routes/routes";
import FormPage from "./pages/FormPage";
import ResultsPage from "./pages/ResultsPage";
import Header from "./components/Header";
import HomePage from "./pages/HomePage";

function App() {
  return (
    <Router>
      <div className="App">
        <Header />
        <div className="container mt-5 d-flex justify-content-center">
          <Switch>
            <Route exact path={ROOT_ROUTE}>
              <HomePage />
            </Route>
            <Route path={FORM_ROUTE}>
              <FormPage />
            </Route>
            <Route path={RESULTS_ROUTE}>
              <ResultsPage />
            </Route>
          </Switch>
        </div>
      </div>
    </Router>
  );
}

export default App;
