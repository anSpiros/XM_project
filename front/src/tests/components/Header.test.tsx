import React from "react";
import Header from "../../components/Header";
import userEvent from "@testing-library/user-event";
import { cleanup, render, screen } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { BrowserRouter, Router } from "react-router-dom";
import App from "../../App";
import { Provider } from "react-redux";
import { createStore } from "redux";
import allReducers from "../../state/reducers";
const store = createStore(allReducers);

describe("Header component", () => {
  afterEach(cleanup);
  test("should render correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("full app rendering/navigating", () => {
    const history = createMemoryHistory();
    render(
      <Provider store={store}>
        <Router history={history}>
          <App />
        </Router>
      </Provider>
    );

    expect(screen.getByText(/Welcome/i)).toBeInTheDocument();

    const leftClick = { button: 0 };
    userEvent.click(screen.getByText(/Form/i), leftClick);

    // check that the content changed to the new page
    expect(screen.getByText(/Email/i)).toBeInTheDocument();
  });
});
