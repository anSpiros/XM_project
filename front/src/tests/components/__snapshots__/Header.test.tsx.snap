// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="container-fluid header"
  >
    <div
      class="row"
    >
      <div
        class="col-12"
      >
        <ul
          class="list-unstyled header-list mb-0"
        >
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/form"
            >
              Form
            </a>
          </li>
          <li>
            <a
              href="/results"
            >
              Results
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
